search/search.py

```diff
@@ -1,15 +1,17 @@
 from functools import reduce
-from typing import List, Union, Iterable
 from operator import or_, add
+from typing import List, Union, Iterable, TypeVar
 
 from django.contrib.postgres.search import TrigramSimilarity
 from django.db import models
 from django.db.models import Q, F, When, Case, Value, BooleanField
 from django.shortcuts import _get_queryset
 
+QuerySet_ = TypeVar('QuerySet_')
```

Comment:
Why do we need these? What is wrong with `django.db.models.QuerySet`?

Reply:
@artemiy312 , we are needed in generic here, because returning type depends on passed type

Added this comment and renamed const:
```
# generic type.
# See code example here to clarify why we are needed generic type:
# https://docs.python.org/3/library/typing.html#generics
QuerySetType = TypeVar('QuerySetType')
```